src/test_typing_extensions.py

```diff
@@ -3591,7 +3591,7 @@ class CP(P[int]):
             self.assertEqual(x.bar, 'abc')
             self.assertEqual(x.x, 1)
             self.assertEqual(x.__dict__, {'foo': 42, 'bar': 'abc'})
-            s = pickle.dumps(P)
+            s = pickle.dumps(GlobalProto)
             D = pickle.loads(s)
```

Comment:
I've renamed this to not have global `P` that is not a ParamSpec. This also caused the wrong inference in the current pyright version mentioned above.

Is it okay to include this in this PR or should it be its own PR?

Reply:
This PR should only contain the changes that are needed to fix #448. There's a couple of reasons for that:
- It makes the PR easier for us to review :-) The diff on GitHub will only show the specific changes that are required to fix the bug in question
- When looking through git history in the future, it will be much easier to see at-a-glance which specific lines needed to be changed in order to fix this bug, and which lines of code preexisted this change

Having said that, I'm not sure pyright complaining about this assignment would be the best reason for changing the name of this variable, so I'm not sure I'd open another PR either. We don't run pyright in CI, and it would be somewhat silly for us to do so in this repository. We have no hope of getting our code to type-check with any type checker, since we are to some extent defining the typing system itself in the code in this repository. If pyright is complaining about this for you locally and it's bugging you, I think the best solution there might be to change your IDE settings so that pyright doesn't run on this repository ;)

Reply:
Follow up question. Some additional artifacts in TypeAliasType.__getitem__, like the dunder attributes or the cases covered by `test_subscription_without_type_params` are not required to fix #448. Should I fork them for another Issue & PR? And if yes, can I base them base them on this PR? I think otherwise it will make things harder (for me).
Possibly I'll try to unify `__getitem__` again and branch based on the version elsewhere.